### PR TITLE
Improve CI workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
"Integrate" as i in "CI".

image

🎁 You have a membership to the Workflow Library of Viktor.
https://github.com/szepeviktor/byte-level-care/tree/master/.github/workflows

### CODE REVIEW

File: .github/workflows/integrate.yml [hunk 1]
```diff
@@ -1,15 +1,16 @@
-name: CI (tests and code style)
+name: Integrate
 
-on: pull_request
+on:
+  pull_request: null
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml:4]: `null` is the base value in YAML

File: .github/workflows/integrate.yml [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
   cs:
-    name: PHP CS Fixer
+    name: Coding Standards
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
+      - name: Checkout repository
+        uses: actions/checkout@v4
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml:13]: v4 just came out

File: .github/workflows/integrate.yml [hunk 3]
```diff
@@ -18,48 +19,48 @@ jobs:
           coverage: none
 
       - name: Install dependencies
-        run: composer install --prefer-dist --no-progress
+        run: composer update --prefer-dist --no-progress
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml:22]: We don't have a lock file!

File: .github/workflows/integrate.yml [hunk 4]
```diff
@@ -18,48 +19,48 @@ jobs:
...
 
       - name: Run PHP CS Fixer
-        run: composer cs
+        run: composer run cs
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml:25]: `composer <command>`
`composer run <script>`

**Comment by otsch** [comment]
@szepeviktor 
I'd take the update to `actions/checkout@v4`, everything else:
* Renaming stuff: I'd like to stick with how things are named currently.
* `pull_request: null` unnecessary
* `composer update` instead of `composer install` => I'd say we don't update, we install for the first time. There's no need for a lock file for install.
* `composer cs` vs `composer run cs` => why? [In the docs](https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands) it's either `composer run-script cs` or `composer cs`, no mention of `composer run cs` and I'd say `composer cs` is just fine 🤷🏻‍♂️

**Comment by szepeviktor** [comment]
All right. I let you bump checkout's version.

**Comment by szepeviktor** [comment]
> * I'd say we don't update, we install for the first time. There's no need for a lock file for install.

Try running `composer install` without a lock file. You get a warning.

**Comment by otsch** [comment]
> Try running `composer install` without a lock file. You get a warning.

Run `composer install` without a lock file and also without a vendor folder and there is no warning. You only get a warning when running it with a vendor folder, containing actual installed packages, but no lock file, which should not occur normally.

**Comment by szepeviktor** [comment]
image

**Comment by otsch** [comment]
> Run composer install without a lock file and also without a vendor folder

So you definitely deleted also the `/vendor` folder before running this? Because I get this warning, when I delete only the lock file and not the vendor folder.

**Comment by szepeviktor** [comment]
> So you definitely deleted also the `/vendor` folder before running this?

Yes.

> Composer version 2.5.5 2023-03-21 11:50:05

**Comment by otsch** [comment]
> > So you definitely deleted also the `/vendor` folder before running this?
> 
> Yes.
> 
> > Composer version 2.5.5 2023-03-21 11:50:05

OK, still...the installation works just fine 🤷🏻‍♂️

**Pull request closed without merging**